zerver/lib/actions.py

```diff
@@ -3497,7 +3497,7 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
 @transaction.atomic
 def do_update_message(user_profile: UserProfile, message: Message, topic_name: Optional[Text],
                       propagate_mode: str, content: Optional[Text],
-                      rendered_content: Optional[Text], prior_mention_user_ids: Set[int],
+                      rendered_content: Text, prior_mention_user_ids: Set[int],
```

Comment:
`rendered_content` should be optional; the structure of this function is that you could have changed either the topic or the content (or both), so we can't assume either is not `None`.  

The correct inference is "content is not None" means "rendered_content is not None".

Reply:
That's fine; this change resolved a `None` value of `rendered_content` being used, so that'll be there for another day instead :)